components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -49,14 +51,36 @@
      * user will assigned to the created role.Internal roles used.
      *
      * @param serviceProvider Service provider
-     * @return application id
      * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
      */
     public void createApplication(ServiceProvider serviceProvider) throws IdentityApplicationManagementException {
 
+        try {
+            addApplication(serviceProvider, null);
+        } catch (IdentityApplicationManagementException idpException) {
+            log.error("Error while creating application: " + serviceProvider.getApplicationName() + " for tenant: " +
+                    getTenantDomain(), idpException);
+            throw idpException;
+        }
+    }
+
+    /**
+     * Creates a service provider with basic information.First we need to create
+     * a role with the
+     * application name. Only the users in this role will be able to edit/update
+     * the application.The
+     * user will assigned to the created role.Internal roles used.
+     *
+     * @param serviceProvider Service provider
+     * @param templateName SP template name
+     * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
+     */
+    public ServiceProvider addApplication(ServiceProvider serviceProvider, String templateName)
```

Comment:
- createApplication()
- addApplication()
service methods does not self explains their distinct duties.

Suggestion,
- createApplication()
- createApplicationWithTemplate()